### PR TITLE
support typescript

### PR DESCRIPTION
**Pull request by 74th**
Thank you for your great works!
I'm using with typescript and create typescript definition.
I would like this bundled with npm if you permit.

### CODE REVIEW

File: tinygradient.d.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+     * Generate gradient with HSVa interpolation
+     * @param {int} steps
+     * @param {Boolean|String} [mode=false]
+     *    - false to step in clockwise
+     *    - true to step in trigonometric order
+     *    - 'short' to use the shortest way
+     *    - 'long' to use the longest way
+     * @return {tinycolor[]}
+     */
+	hsv(steps: number, mode: boolean): tinycolorInstance[];
```
Comment (mistic100) [inline tinygradient.d.ts]: you would need a typedef here

```ts
type ArcMode = boolean | 'short' | 'long';
```

File: tinygradient.d.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/*!
+ * TinyGradient 0.4.0
+ * Copyright 2014-2018 Damien "Mistic" Sorel (http://www.strangeplanet.fr)
+ * Licensed under MIT (http://opensource.org/licenses/MIT)
+ */
+
+declare var tinygradient: tinygradient;
+
+interface TinyGradient {
```
Comment (mistic100) [inline tinygradient.d.ts]: doesn't the interface needs to be exported (as well as the types) ?

Comment (74th) [reply tinygradient.d.ts]: Yes, my product has belong code.

```
interface gradationSetting {
	high_filter: number,
	low_filter: number,
	high_color: string,
	low_color: string,
	grad?: TinyGradient;
};
const calcGradation = (p: number, setting: gradationSetting): string => {
	if (setting.grad === undefined) {
		setting.grad = tinygradient([setting.low_color, setting.high_color]);
	}
	...
}
```

File: tinygradient.d.ts (hunk 1 above)
Comment (74th) [reply tinygradient.d.ts]: I fix it.

```
hsv(steps: number, mode: ArcMode): tinycolorInstance[];
```

And also

```
css(mode?: 'linear' | 'radial', direction?: string): string;
```

**Comment by 74th** [comment]
Thanks for your review.
I fixed them.

File: tinygradient.d.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    hsvAt(pos: number): tinycolorInstance;
+
+}
+
+interface tinygradient {
+    /**
+     * @class tinygradient
+     * @param {*} stops
+     */
+    (stops: TinyGradientSteps[]): TinyGradient;
```
Comment (mistic100) [inline tinygradient.d.ts]: can't you use ColorInput type from tinycolor here ?

File: tinygradient.d.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+}
```
Comment (mistic100) [inline tinygradient.d.ts]: if you want to be exhaustive, it's missing the 5 static methods (they are created lines 384-399) of the JS file.

```ts
    /**
     * Generate gradient with RGBa interpolation
     * @param {*} stops
     * @param {int} steps
     * @return {tinycolor[]}
     */
    rgb(stops: TinyGradientSteps[], steps: number): tinycolorInstance[];

    /**
     * Generate gradient with HSVa interpolation
     * @param {*} stops
     * @param {int} steps
     * @param {Boolean|String} [mode=false]
     *    - false to step in clockwise
     *    - true to step in trigonometric order
     *    - 'short' to use the shortest way
     *    - 'long' to use the longest way
     * @return {tinycolor[]}
     */
    hsv(stops: TinyGradientSteps[], steps: number, mode: ArcMode): tinycolorInstance[];

    /**
     * Generate CSS3 command (no prefix) for this gradient
     * @param {*} stops
     * @param {String} [mode=linear] - 'linear' or 'radial'
     * @param {String} [direction] - default is 'to right' or 'ellipse at center'
     * @return {String}
     */
    css(stops: TinyGradientSteps[], mode?: 'linear' | 'radial', direction?: string): string;

    /**
     * Returns the color at specific position with RGBa interpolation
     * @param {*} stops
     * @param {float} pos, between 0 and 1
     * @return {tinycolor}
     */
    rgbAt(stops: TinyGradientSteps[], pos: number): tinycolorInstance;

    /**
     * Returns the color at specific position with HSVa interpolation
     * @param {*} stops
     * @param {float} pos, between 0 and 1
     * @return {tinycolor}
     */
    hsvAt(stops: TinyGradientSteps[], pos: number): tinycolorInstance;
```

Comment (74th) [reply tinygradient.d.ts]: I'm surprised to know that such great tricks existed!
I added them.

File: tinygradient.d.ts (hunk 3 above)
Comment (74th) [reply tinygradient.d.ts]: I think that's right. I fixed it.

**Comment by mistic100** [comment]
Please remove the package lock

**Comment by Flo5k5** [comment]
Hi, is it possible to push the changes on the npm repository please?

**Comment by mistic100** [comment]
Yeah, just needs a update because the signature recently changed https://github.com/DefinitelyTyped/DefinitelyTyped/commit/080c946cb2426828a19db4605b5fb2b07e661fb1#diff-8d722423c86bf59d286e58cc16427244 (on a patch version !)

**Comment by Flo5k5** [comment]
Thanks for the update ;)

**Pull request merged**